util/version/version.go

```diff
@@ -201,12 +201,21 @@ func WithBuildTags() CompareOption {
 	}
 }
 
+func WithoutPreReleases() CompareOption {
+	return func(c *comparer) {
+		c.buildTags = false
+	}
+}
+
 // Compare 2 semver versions.
 // Defaults to doing the standard semver comparison when no options are specified.
 // The comparison logic can be modified by passing additional compare options.
 // Example: using the WithBuildTags() option modifies the compare logic to also
 // consider build tags when comparing versions.
 func Compare(a, b semver.Version, options ...CompareOption) int {
+	a.Pre = nil
+	b.Pre = nil
```

Comment:
```suggestion
	if c. withoutPreReleases {
		a.Pre = nil
		b.Pre = nil
	}
```

And let's move it below the code where we collect the options (so ~ l.223)